File: app/src/main/java/com/vikingsen/cheesedemo/ux/cheeselist/CheeseListModule.java
package com.vikingsen.cheesedemo.ux.cheeselist;

import dagger.Module;
import dagger.Provides;

@Module
public class CheeseListModule {
    private final CheeseListContract.View view;

    CheeseListModule(CheeseListContract.View view) {
        this.view = view;
    }

    @Provides
    CheeseListContract.View provideView() {
        return view;
    }
}

File: app/src/main/java/com/vikingsen/cheesedemo/ux/cheeselist/CheeseListModule.kt
package com.vikingsen.cheesedemo.ux.cheeselist

import dagger.Module
import dagger.Provides

@Module
class CheeseListModule(private val view: CheeseListContract.View) {

    @Provides
    internal fun provideView(): CheeseListContract.View {
        return view
    }
}
